### PR TITLE
Merge in MicroPython 1.13

### PR DESCRIPTION
**Pull request by tannewt**
Progress on #2999

### CODE REVIEW

File: docs/library/errno.rst [hunk 1]
```diff
@@ -1,32 +1,34 @@
-:mod:`errno` -- system error codes
+:mod:`uerrno` -- system error codes
 ===================================
 
-.. module:: errno
+.. module:: uerrno
    :synopsis: system error codes
 
-|see_cpython_module| :mod:`cpython:errno`.
+|see_cpython_module| :mod:`python:errno`.
```
[the remark this answers is not in the document]
Comment (tannewt) [reply docs/library/errno.rst:7]: Our RTD is setup to be `cpython`. (See intersphinx mapping in `conf.py`.)

File: py/objint_mpz.c [hunk 2]
```diff
@@ -436,11 +438,27 @@ mp_int_t mp_obj_int_get_checked(mp_const_obj_t self_in) {
             return value;
         } else {
             // overflow
-            mp_raise_msg(&mp_type_OverflowError, translate("overflow converting long int to machine word"));
+            mp_raise_msg(&mp_type_OverflowError, MP_ERROR_TEXT("overflow converting long int to machine word"));
         }
     }
 }
 
+mp_uint_t mp_obj_int_get_uint_checked(mp_const_obj_t self_in) {
```
Comment (dhalbert) [inline py/objint_mpz.c:446]: There are places we can this `mp_obj_int_get_uint_checked()`

File: py/objtuple.c [hunk 3]
```diff
@@ -37,6 +37,9 @@
 // type check is done on getiter method to allow tuple, namedtuple, attrtuple
 #define mp_obj_is_tuple_compatible(o) (mp_obj_get_type(o)->getiter == mp_obj_tuple_getiter)
 
+// type check is done on getiter method to allow tuple, namedtuple, attrtuple
+#define mp_obj_is_tuple_compatible(o) (mp_obj_get_type(o)->getiter == mp_obj_tuple_getiter)
+
```
Comment (dhalbert) [inline py/objtuple.c:42]: This looks like a duplicated definition??

**Review by dhalbert — CHANGES_REQUESTED** [review]
Ignore the MP_ERROR_TEXT comment; I can't find it in the sea of files.

Definitely getting simpler. Thanks for continuing to do this!

File: extmod/modubinascii.c [hunk 4]
```diff
@@ -9,17 +9,16 @@
 
 #include "py/runtime.h"
 #include "py/binary.h"
-#include "extmod/modubinascii.h"
 
 static void check_not_unicode(const mp_obj_t arg) {
     #if MICROPY_CPYTHON_COMPAT
     if (mp_obj_is_str(arg)) {
-        mp_raise_TypeError(translate("a bytes-like object is required"));
+        mp_raise_TypeError(MP_ERROR_TEXT("a bytes-like object is required"));
```
[the remark this answers is not in the document]
Comment (dhalbert) [reply extmod/modubinascii.c:16]: Never mind. Found it way down below.

**Comment by tannewt** [comment]
> Ignore the MP_ERROR_TEXT comment; I can't find it in the sea of files.
> 
> Definitely getting simpler. Thanks for continuing to do this!

For the record MP_ERROR_TEXT is here for CP: https://github.com/tannewt/circuitpython/blob/merge_1.13/supervisor/shared/translate.h#L84 (I deleted the MP version.)

File: py/objint_mpz.c (hunk 2 above)
Comment (tannewt) [reply py/objint_mpz.c:446]: Where we can use this? I can't think of any immediately. Can we do that in a follow up?

Comment (dhalbert) [reply py/objint_mpz.c:446]: Oh, I don't mean we have to use this now. But I recently was working on som.e code that required positive arguments, and to have this checked automatically would be nice

**Review by dhalbert — APPROVED** [review]
Everything I had has been addressed. Thanks for taking on this! Onward!

**Pull request merged**